### ISSUE TITLE
You must explicitly open the connection before executing anything when passing in a DbConnection to Database

### ISSUE THREAD

**Issue by dgulliver** [schotime/NPoco#436]
Hi I'm using mySQL as my Database but when I execute the Insert I get the error:

> You must explicitly open the connection before executing anything when passing in a DbConnection to Database

Table Class:
```
`namespace NPOCO_TEST
{
    [TableName("tblDiary")]
    [PrimaryKey("DI_ID")]
    public class Diary
    {
        public int DI_ID { get; set; }
        public int DI_DT_ID { get; set; }
        public int DT_USER_ID { get; set; }
        public bool DI_CONFIRMED { get; set; }
        public bool DI_DONE { get; set; }
        public int DI_VH_ID { get; set; }
        public int DI_OL_ID { get; set; }
        public int DI_BP_ID { get; set; }
        public int DI_AG_ID { get; set; }
        public int DI_SM_ID { get; set; }
        public int DI_NUMBER_1 { get; set; }
        public int DI_NUMBER_2 { get; set; }
        public int DI_PH_ID { get; set; }
        public string DI_EMAIL_TO { get; set; }
        public string DI_EMAIL_CC { get; set; }
        public string DI_SUBJECT { get; set; }
        public string DI_TEXT { get; set; }
        public string DI_ATTACHMENTS { get; set; }
        public string DI_FROM { get; set; }
        public string DI_TO { get; set; }
        public DateTime DI_DATE { get; set; }
        public DateTime DI_TIME { get; set; }
        public DateTime DI_DATE_CHANGED { get; set; }
    }
}`
```

Insert Code:
```
public void AddDiary_NPOCO()
        {
            MySqlConnection conn = new MySqlConnection(ConfigurationManager.ConnectionStrings["TEST"].ConnectionString);

            Diary u = new Diary()
            {
                
                DI_DATE = DateTime.Today,
                DI_DATE_CHANGED = DateTime.Now,
                DI_DT_ID = 6,
                DI_DONE = false,
                DI_EMAIL_TO = "Darren.12345@gmail.com",
                DI_EMAIL_CC = "darren1234@gmail.com",
                DI_SUBJECT = "Hello World",
                DI_TEXT = "x",


            };

           
            using (IDatabase db = new Database(conn))
            {
                db.Insert(u);
            }
            
        }
```
The error happens on the db.Insert(u); line.

NPoco Ver:3.9.2
mySQL .Net Connector:6.9.8

I am new to NPoco and C# with VB.Net experience so I'm probably missing something silly, But the Using statement should open and dispose  of the connection.

Cheers
Darren

**Comment by schotime** [comment]
If you pass in the connection object you are required to open it. Or call open shared connection. Npoco will not manage an external connection. If you just pass the connection string then it will manage it.

**Comment by dgulliver** [comment]
Thanks for explaining. 
I did use the  connection string but got a different error. I will add when I get to work in the morning.

Thanks Darren

**Comment by dgulliver** [comment]
I found the problem. I should of passed in the connection string name which the NPoco Database class looks up in the App.Config. The other thing I found that the connection string needs **providerName="MySql.Data.MySqlClient"** otherwise I get another error about not recognising the Port attribute.

So now this part of the code works because I have a Connection String called TEST in my App.Config.

```
using (IDatabase db = new Database("TEST"))
            {
                db.Insert(u);
            }
```
Thanks again,
Darren
